server/services/message-service.js

```diff
@@ -1,32 +1,53 @@
-const amqp = require('amqplib/callback_api')
+const rheapromise = require('rhea-promise')
 const config = require('../config')
+const calculationQueue = 'calculation'
+const scheduleQueue = 'schedule'
 
 module.exports = {
-  publishClaim: function (claim) {
-    try {
-      const messageQueue = config.messageQueue
-      amqp.connect(messageQueue, function (err, conn) {
-        if (err) {
-          console.log(err)
-        }
-        conn.createChannel(function (err, ch) {
-          if (err) {
-            console.log(err)
-          }
 
-          const data = JSON.stringify(claim)
+  publishClaim: async function (claim, messageQueueOptions) {
+    let configureMQ = function (options) {
+      return {
+        transport: options.transport,
+        port: options.port,
+        reconnect_limit: 10,
+        host: options.address,
+        hostname: options.address,
+        username: options.user,
+        password: options.pass
+      }
+    }
+    let sendClaim = async function (claim, connection, queueName) {
+      const data = JSON.stringify(claim)
+      const queueOptions = { target: { address: queueName } }
+      const sender = await connection.createSender(queueOptions)
+      let delivery
+      try {
+        console.log(`Sending claim to ${queueName}`)
+        delivery = await sender.send({ body: data })
+      } catch (error) {
+        throw error
+      }
+      await sender.close()
+      return delivery
+    }
+
+    try {
+      const connectionOptions = configureMQ(messageQueueOptions || config.messageQueue)
+      console.log(connectionOptions)
 
-          const calculationQueue = 'calculation'
-          ch.assertQueue(calculationQueue, { durable: false })
-          ch.sendToQueue(calculationQueue, Buffer.from(data))
-          console.log('claim queued for calculation')
+      const connection = new rheapromise.Connection(connectionOptions)
+      console.log(`New claim to send to message queues : ${claim}`)
+      try {
+        await connection.open()
 
-          const scheduleQueue = 'schedule'
-          ch.assertQueue(scheduleQueue, { durable: false })
-          ch.sendToQueue(scheduleQueue, Buffer.from(data))
-          console.log('claim queued for scheduling')
-        })
-      })
+        const delivery = await Promise.all([sendClaim(claim, connection, calculationQueue),
+          sendClaim(claim, connection, scheduleQueue)])

```

Comment:
I'd be tempted to restructure lines 44-45 as
```
const delivery = await Promise.all([
  sendClaim(claim, connection, calculationQueue),
  sendClaim(claim, connection, scheduleQueue)
])
```
for readability

Reply:
Yes, that's more readable.